src/Form/Type/ToolUpdateType.php

```diff
@@ -23,6 +23,30 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                     'label'        => false,
                     'by_reference' => false,
                 ]
+            )
+            ->add(
+                'params',
+                CollectionType::class,
+                [
+                    'required'     => false,
+                    'entry_type'   => ParamType::class,
+                    'allow_add'    => true,
+                    'allow_delete' => true,
+                    'label'        => false,
+                    'by_reference' => false,
+                ]
+            )
+            ->add(
+                'logs',
+                CollectionType::class,
+                [
+                    'required'     => false,
+                    'entry_type'   => LogType::class,
+                    'allow_add'    => true,
+                    'allow_delete' => true,
+                    'label'        => false,
+                    'by_reference' => false,
+                ]
```

Comment:
No need to add these fields here, they are inherited from ToolType. Also missing prototype options, so in update form if multiple params/logs are added only last one is saved.

Reply:
Bets would be to just remove from ToolUpdateType